### DISCUSSION TITLE
[ Discord Notification ] Replace plain text notifications to Embeds

### DISCUSSION THREAD

**Discussion by ShadowArcanist** [coollabsio/coolify#3540] [Improvement Requests]
At the moment Coolify sends discord notification via webhook as a plain text message, it would be great if the messages were sent as a Embed.

Also it would be alot helpful if users can enter a role id or user id to get pinged when some critical notifications were sent (example:  Database backup failed)

I attached screenshots of some of the ideas I had on mind:

Color codes used on the images:

Green - #a1ffa5
Red - #ff705f
Orange/Yellow - #ffa743

![image](https://github.com/user-attachments/assets/3a2f069c-c397-4ee7-8451-84fde7ff4bda)

![image](https://github.com/user-attachments/assets/e052dd06-280a-4a01-9b0d-4a334e117fa7)

![image](https://github.com/user-attachments/assets/f37f4dcf-36fe-48bb-9cc7-1053c945104d)

![image](https://github.com/user-attachments/assets/ad4b0058-e88a-4f8c-a5ae-b13451f2ce79)

![image](https://github.com/user-attachments/assets/136a5ec2-6fc6-4b0e-bd6d-8b28a8101350)

![image](https://github.com/user-attachments/assets/237e0bdf-fbf4-4dcd-9881-0d6013f2b258)

![image](https://github.com/user-attachments/assets/20fb6cf1-c2d3-4bef-b937-86151ff4b067)

![image](https://github.com/user-attachments/assets/62c2507b-c84b-4912-ab3e-3d4e2597f17f)

![image](https://github.com/user-attachments/assets/ea293e67-450c-4e8c-9740-4802eb22669b)

![image](https://github.com/user-attachments/assets/829728be-8106-4e10-ac7d-fb8fa10c0967)

![image](https://github.com/user-attachments/assets/45a4a86f-bfa1-4f61-a2ca-548b9909d5e7)

![image](https://github.com/user-attachments/assets/f401dd5b-7611-42df-bfe7-52f5cc3f15ce)

![image](https://github.com/user-attachments/assets/d6d594bf-eb7e-4662-ad27-b0004108fcce)

**Comment by M1ck0** [comment]
+1 for this

**Comment by Vahor** [comment]
Maybe out of scope but it would be nice if we were able to put our own json template for the messages, and coolify would only replace "variables" like time/url/servername/message etc.
The suggested embed being the default values.

This could help for translations, style and custom ping

**Comment by graphiostudio** [comment]
Yes, please :) This would be amazing.

**Comment by jnovakdev** [comment]
This seems like a fine first contribution :)

I've prepared the logic, check it out: https://github.com/coollabsio/coolify/compare/main...jnovakdev:coolify:main
For critical notifications (deploy failed, db backup failed, etc.) I've added `isCritical` property to tag everyone in the channel.

I will discuss this with Andras on the stream if the code style is something he is fine with and will finish up the rest of the notifications (currently I've set only the Test one)

Not sure about the colors though, the red(danger) might be worth updating. 

![Screenshot 2024-09-30 at 0 43 08](https://github.com/user-attachments/assets/295022f7-9af9-4fc4-9864-7b80419cc43a)

**Reply by DysektAI** [reply]
Looking good! I am really liking the embeds better this way, much cleaner and obvious display. Look forward to this being added would 100% love there to be a "Default" embed layout/theme, but have the ability to customize it.

As mentioned originally by @Vahor being able to have "Advanced/Custom" layouts/themes of our own would be a super nice addition.

I do believe some of the fields should be [Default if not set] like the obvious Embed message requirements like "Title" and "Color".

For now though this is 100% a great starting point.

**Comment by 7oque** [comment]
Heya this would be appreciated as at the moment the default behaviour has an `@here` which is not easy to configure (as the notifications come from a webhook and do not have a discord role)

https://cdn.discordapp.com/attachments/1009177753230245928/1318911255854972938/image.png?ex=67640b53&is=6762b9d3&hm=6aa77007db85aa9dea637c2cb4fc31439ee56ba60a760b47d0d7401cf6027875&

Thanks so much 🤘
